tools/setup-dev.sh

```diff
@@ -318,9 +310,11 @@ if [ -d venv ] ; then
 fi
 
 
-# we need system site packages because librados python bindings appear to only
-# be available as a package. It might be a good idea to compile it from the
-# ceph repo we keep as a submodule, but it might be overkill at the moment?
+# We need system site packages because librados python bindings are only
+# available as a package, they're not on pypi (and Tim thinks they probably
+# never will be). Setting --system-site-packages here means the venv will
+# work properly if someone uses it inside the image, where we need to be
+# able to fall back to system packages to get librados.
 $PYTHON -m venv --system-site-packages venv || exit 1
```

Comment:
I really don't understand how host's venv can help on the guest VM, why do I need this.

Reply:
This is a bit obscure... If we create the venv with `--system-site-packages`, it sets `include-system-site-packages = true` inside `venv/pyvenv.cfg`, so, later, if you _use_ the venv from inside the image, this means it will correctly pick up python3-rados from the system packages inside the image. If we created the venv without that option, using the venv inside the image wouldn't work.

That said, IMO you shouldn't usually need to use the venv inside the image at all (because everything is installed system wide already), so it's a bit of a moot point, but, when chatting to @votdev the other day, he said he _does_ use the venv inside the image sometimes, to handle added dependencies easily, without having to rebuild the image.

Reply:
I see your point, but in general we MUST NOT use the venv backed on the host inside virtual images. So if one wants to have venv inside the image it should be created right in the guest system in order to avoid any kind of misfit of packages and modules.

Reply:
Right, makes sense.

Reply:
If we want to force setup of a proper inside-image venv, then I think we should do that in run_aquarium.sh, which we can do in a separate PR.